tests/queries/0_stateless/01156_pcg_deserialization.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+declare -a engines=("Memory" "MergeTree order by n" "Log")
+
+for engine in "${engines[@]}"
+do
+  $CLICKHOUSE_CLIENT -q "drop table if exists t";
+  $CLICKHOUSE_CLIENT -q "create table t (n UInt8, a1 AggregateFunction(groupArraySample(1), UInt8)) engine=$engine"
+  $CLICKHOUSE_CLIENT -q "insert into t select number % 5 as n, groupArraySampleState(1)(toUInt8(number)) from numbers(10) group by n"
+
+  $CLICKHOUSE_CLIENT -q "select * from t format TSV" | $CLICKHOUSE_CLIENT -q "insert into t format TSV"
+  $CLICKHOUSE_CLIENT -q "select countDistinct(n), countDistinct(a1) from t"
+
```

Comment:
It used to produce different results with `Memory` and `MergeTree`/`Log`, I have no ideas why:
https://gist.github.com/tavplubix/8fafec4bca6a1b78087afd99aea85ff9

Reply:
You did not specify seed, so random seed is selected.